bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Election.cs

```diff
@@ -112,29 +113,74 @@ public unsafe ElementModQ CryptoExtendedBaseHash
-            if (status != Status.ELECTIONGUARD_STATUS_SUCCESS)
-            {
-                Console.WriteLine($"CiphertextElectionContext Error Status: {status}");
-            }
+            status.ThrowIfError();
         }
 
+        /// <summary>
+        ///  Makes a CiphertextElectionContext object.
+        ///
+        /// <param name="number_of_guardians"> The number of guardians necessary to generate the public key </param>
+        /// <param name="quorum"> The quorum of guardians necessary to decrypt an election.  Must be less than `number_of_guardians` </param>
+        /// <param name="publicKey"> the public key of the election </param>
+        /// <param name="commitmentHash"> the hash of the commitments the guardians make to each other </param>
+        /// <param name="manifestHash"> the hash of the election metadata </param>
+        /// </summary>
         public unsafe CiphertextElectionContext(ulong numberOfGuardians,
                 ulong quorum,
                 ElementModP publicKey,
                 ElementModQ commitmentHash,
                 ElementModQ manifestHash)
         {
             var status = NativeInterface.CiphertextElectionContext.Make(
-                numberOfGuardians, quorum, publicKey.Handle, commitmentHash.Handle, manifestHash.Handle, out Handle);
-            if (status != Status.ELECTIONGUARD_STATUS_SUCCESS)
-            {
-                Console.WriteLine($"CiphertextElectionContext Error Status: {status}");
-            }
+                numberOfGuardians, quorum, publicKey.Handle,
+                commitmentHash.Handle, manifestHash.Handle, out Handle);
+            status.ThrowIfError();
+        }
+
+        /// <summary>
+        ///  Makes a CiphertextElectionContext object.
+        ///
+        /// <param name="numberOfGuardians"> The number of guardians necessary to generate the public key </param>
+        /// <param name="quorum"> The quorum of guardians necessary to decrypt an election.  Must be less than `number_of_guardians` </param>
+        /// <param name="publicKey"> the public key of the election </param>
+        /// <param name="commitmentHash"> the hash of the commitments the guardians make to each other </param>
+        /// <param name="manifestHash"> the hash of the election metadata </param>
+        /// <param name="extendedData"> an unordered map of key value strings revelant to the consuming application </param>
+        /// </summary>
+        public unsafe CiphertextElectionContext(ulong numberOfGuardians,
+                ulong quorum,
+                ElementModP publicKey,
+                ElementModQ commitmentHash,
+                ElementModQ manifestHash,
+                LinkedList extendedData)
+        {
+            var status = NativeInterface.CiphertextElectionContext.Make(
+                numberOfGuardians, quorum, publicKey.Handle,
+                commitmentHash.Handle, manifestHash.Handle, extendedData.Handle, out Handle);
```

Comment:
notice we pass in the C# managed types into the function parameters but then we use the internal access of the Handle to call the native methods. that gives us some semblance of type safety even though they are just `IntPtr` types under the hood.

Reply:
I forget, do you have to [pin](https://docs.microsoft.com/en-us/dotnet/framework/interop/copying-and-pinning) C# managed types with `GCHandle.FromIntPtr()` when you pass them to native methods so the memory isn't moved by the GC while in native code?  I know strings are handled automatically, but it's been a while since I've done any of this.  Probably a dumb question, feel free to ignore.

Reply:
yes, that's one of the things the `SafeHandle` base class does under the hood iirc. check out the [ElectionguardSafeHandle](https://github.com/microsoft/electionguard-cpp/blob/main/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/ElectionguardSafeHandle.cs) subclass and the `ReleaseHandle` override